### PR TITLE
Switch from npm to yarn

### LINKED ISSUE #97: Shift from npm to yarn
## Description

I prefer (for good reasons) to use yarn over npm. Yarn is faster and more secure than npm, and also easier to type (you don't need the run and than command anymore)

## Requirements

- [ ] Add yarn

### PR DESCRIPTION
**Pull request by Anissa3005**
# Description

Previously we used npm as our package manager but after using yarn on other projects I find yarn faster and easier to work with. 

We will work with the newest stable version of Yarn and installing is quite simple. The only thing you have to do is enable `corepack` and you can start using yarn. https://yarnpkg.com/getting-started/install

This is something new so I don't have experience doing it that way, so please test it out using this branch and see if you can run the yarn command. For example `yarn lint` .

This is what will change:

`npm run <command>`  => `yarn <command>`
`npm install` => `yarn OR yarn install`
`npm install` <package> => `yarn add <package>`
`npm uninstall` <package> => `yarn remove <package>`

# Closes issue(s)

Resolves #97 

# How to test

Checkout this branch and try to run a yarn commands.

# Changes include

1. Add yarn lock file
2. Change npm to yarn
3. Add engines to package.json so we run the same yarn and node version

# Checklist

- [x] I have tested this code
- [x] I have updated the README

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,33 +9,32 @@
...
   },
-  "husky": {
-    "hooks": {
-      "pre-push": ".husky/pre-push"
-    }
-  },
   "keywords": [],
   "author": "",
   "license": "ISC",
   "dependencies": {
```
Comment (shonancanuck) [inline package.json:36]: During install I get this message:
```
YN0000: ┌ Post-resolution validation
➤ YN0002: │ backend@workspace:. doesn't provide @types/node (pdeab6), requested by ts-node.
➤ YN0086: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code.
➤ YN0000: └ Completed
```
adding `@types/node` seems to remove the warning, although we may need to specify v18.

**Review by shonancanuck — DISMISSED** [review]
Looks good, and seems to work except for the seed script.

I did have some issues exporting from `@prisma/client` after the initial install, but after inspecting the files in node modules it resolved itself. I also got the warning about dependencies during install, but that looks like an easy fix.

File: package.json (hunk 1 above)
Comment (Anissa3005) [reply package.json:36]: This can be ignored for now
This has nothing to do with yarn.

Comment (Anissa3005) [reply package.json:36]: I have added `@types/node` dev dep

**Review by oddpharmacy — COMMENTED** [review]
Lgtm as well~ 
`yarn lint` and `yarn install` work fine for me.

**Review by oddpharmacy — APPROVED** [review]
lgtm!

**Pull request merged**